indico/modules/events/api.py

```diff
@@ -255,8 +273,243 @@ def _iterate_objs(objIds):
+        }
+
+    def _serialize_session(self, session):
+        return {
+            'folders': build_folders_api_data(session),
+            # TODO: Add when #2272 gets merged
+            # 'startDate': self._serialize_date(session.start_dt) if session.blocks else None,
+            # 'endDate': self._serialize_date(session.end_dt) if session.blocks else None,
+            '_type': 'Session',
+            'sessionConveners': [self._serialize_convener(convener) for convener in session.conveners],
+            'title': session.title,
+            'color': '#{}'.format(session.colors.background),
+            'textColor': '#{}'.format(session.colors.text),
+            'description': session.description,
+            'material': build_material_legacy_api_data(session),
+            'isPoster': session.is_poster,
+            'url': url_for('event.sessionDisplay', confId=session.event_id, sessionId=session.friendly_id,
+                           _external=True),  # TODO: Change to new endpoint once it's there
+            'protectionURL': url_for('sessions.session_protection', session.event_new, session, _external=True),
+            'roomFullname': session.room_name,
+            'location': session.venue_name,
+            'address': session.address,
+            '_fossil': 'sessionMinimal',
+            'numSlots': len(session.blocks),
+            'id': session.friendly_id,
+            'room': session.room.name if session.room else session.room_name
+        }
+
+    fossils_mapping = {
+        'event': {
+            None: 'conferenceMetadata',
+            'contributions': 'conferenceMetadataWithContribs',
+            'subcontributions': 'conferenceMetadataWithSubContribs',
+            'sessions': 'conferenceMetadataWithSessions'
+        },
+        'contribution': {
+            'contributions': 'contributionMetadata',
+            'subcontributions': 'contributionMetadataWithSubContribs',
+            'sessions': 'contributionMetadataWithSubContribs'
+        },
+        'person': {
+            'Avatar': 'conferenceChairMetadata',
+            'ConferenceChair': 'conferenceChairMetadata',
+            'ContributionParticipation': 'contributionParticipationMetadata',
+            'SubContribParticipation': 'contributionParticipationMetadata'
+        }
+    }
+
+    def _build_event_basic_api_data(self, event):
+        return {
+            '_type': 'Conference',
+            'id': event.id,
+            'title': event.title,
+            'description': event.description,
+            'startDate': self._serialize_date(event.start_dt),
+            'timezone': event.timezone,
+            'endDate': self._serialize_date(event.end_dt),
+            'room': event.room_name,
+            'location': event.venue_name,
+            'address': event.address,
+            'type': event.type
+        }
+
+    def _build_event_api_data(self, event):
+        self._detail_level = get_query_parameter(request.args.to_dict(), ['d', 'detail'])
```

Comment:
i don't think this is a good place to set this. it's the same for each event anyway